flink-core/src/main/java/org/apache/flink/api/common/PlanExecutor.java

```diff
@@ -129,6 +129,8 @@ public boolean isPrintingStatusDuringExecution() {
 	 */
 	public abstract String getOptimizerPlanAsJSON(Plan plan) throws Exception;
 
+	public abstract String getOptimizerPlanContext(Plan plan, boolean extended) throws Exception;
```

Comment:
If we reuse the JSON string (as proposed in my comment at `ExecutionEnvironment`) we do not need this method either.